### ISSUE TITLE
Overriding the Sampled Value for an Optimized Sampler

### ISSUE THREAD

**Issue by pford221** [optuna/optuna#1973]
Hello. Great library and framework. Thanks so much for your contribution to more flexible hyperparameter search.

I have a quick question I couldn't find the answer to elsewhere. 

Suppose I'm using a sampler such as `TPESampler` and the sampler suggests a value during a trial for a parameter. If a user then manually overrides the value of the suggested parameter with a hard-coded value, will the sampler's future suggestions be influenced by the originally suggested sample value (that was overriden) or by the manually overriden value that actually went into the function-to-be-optimized? 

To make it concrete. In LightGBM there are several `boosting_type`s. One of those is "goss" or gradient one-sided sampling. When `boosting_type == "goss"` the `bagging_fraction` parameter must be equal to 1.0. So in my code, if the `TPESampler` samples `boosting_type = "goss"` and samples `bagging_fraction < 1` then I go in and manually override the sampled `bagging_fraction` by setting it equal to 1.0 (i.e. `params["bagging_fraction"] = 1.0`). So my question is if the `TPESampler` will evaluate the last set of hyperparamters (in conjunction with the loss) at the originally sampled `bagging_fraction` or at 1.0 when deciding what space to sample from next?

Thanks in advance for your help!

**Comment by HideakiImamura** [comment]
Hi @pford221. For your question, samplers of Optuna assume that the originally suggested sampled value is actually used to calculate the objective function. Therefore, your overridden value is ignored by samplers.

However, Optuna supports the conditional search space. How about the following?

```python
def objective(trial):
    boosting_type = trial.suggest_categorical("boosting_type", ["goss", ...])

    if boosting_type == "goss":
        bagging_fraction = 1.0
    else:
        bnagging_fraction = trial.suggest_float("bagging_fraction", 0., 1.)
    
   ...
```

**Comment by pford221** [comment]
Thank you @HideakiImamura. This makes sense and I'm glad I checked. In most cases I would do exactly what you suggest, but in this specific case I'm working with a potentially unordered dictionary of parameter names and suggested sampled ranges that is being passed from a higher-level function, and so I can't guarantee that `boosting_type` will be sampled before `bagging_fraction` due to the dictionary order. I will have to adjust the logic to ensure this is the case. Many thanks!

**Comment by HideakiImamura** [comment]
Thanks, @pford221! So, let me close this issue?

**Comment by pford221** [comment]
Yes, please do.
